sechub-developertools/src/main/java/com/mercedesbenz/sechub/developertools/pds/PDSSolutionTestFilesGenerator.java

```diff
@@ -0,0 +1,276 @@
+            }
+
+            SecHubFileSystemConfiguration fileSystem = fileSystemOpt.get();
+            copyFileOrFolder(baseTarget, fileSystem.getFiles(), false);
+            copyFileOrFolder(baseTarget, fileSystem.getFolders(), true);
+
+        }
+    }
+
+    private void copyFileOrFolder(File baseTarget, List<String> files, boolean targetIsDirectory) throws FileNotFoundException, IOException {
+        for (String fileName : files) {
+            File file = new File(originConfigFile.getParentFile(), fileName);
+            boolean notExisting = !file.exists();
+
+            File targetFile = new File(baseTarget, fileName);
+            if (targetIsDirectory) {
+                targetFile.mkdirs();
+            } else {
+                targetFile.getParentFile().mkdirs();
+            }
+            if (notExisting) {
+                if (createPseudoFilesWhenNotExisting) {
+                    if (targetIsDirectory) {
+                        File pseudoFile = new File(targetFile, "gen_placeholder_because_origin_not_existing.txt");
+                        pseudoFile.createNewFile();
+                        getOutputHandler().output("WARN: origin folder:" + file.getAbsolutePath() + " did not exist. So created folder at "
+                                + targetFile.getAbsolutePath() + " with place holder file.");
+
+                    } else {
+                        targetFile.createNewFile();
+                        getOutputHandler().output(
+                                "WARN: origin file:" + file.getAbsolutePath() + " did not exist. So created pseudo file:" + targetFile.getAbsolutePath());
+                    }
+                    // we cannot create but have created pseudo-files/empty directories, so stop
+                    // here
+                    continue;
+                } else {
+                    throw new FileNotFoundException("Cannot copy file/folder because it does not exist:" + file.getAbsolutePath());
+                }
+
+            }
+            if (targetIsDirectory) {
+                FileUtils.copyDirectory(file, targetFile);
+            } else {
+                FileUtils.copyFile(file, targetFile);
+            }
+        }
+    }
+
+    private String writeReducedConfigFile() throws IOException {
+        String recucedSecHubConfigJson = SecHubConfigurationModelReducedCloningSupport.DEFAULT.createReducedScanConfigurationCloneJSON(config, scanType);
+        SecHubConfigurationModel reducedConfig = SecHubConfiguration.createFromJSON(recucedSecHubConfigJson);
+        recucedSecHubConfigJson = JSONConverter.get().toJSON(reducedConfig, true);
+
+        File reducedSecHubConfigFile = new File(tmpFolder, "reducedSecHubJson_for_" + scanType.getId() + ".json");
+        writer.save(reducedSecHubConfigFile, recucedSecHubConfigJson, true);
+        return recucedSecHubConfigJson;
+    }
+
+    private void writePDSJobDataFile(String recucedSecHubConfigJson) throws AdapterException, IOException {
```

Comment:
```suggestion
    private void writePDSJobDataFile(String redcucedSecHubConfigJson) throws AdapterException, IOException {
```

Reply:
will change to `reducedSecHubConfigJson` and not to `redcucedSecHubConfigJson` ... but I get the point  ;-)